src/test/java/StringManipulationTest.java

```diff
@@ -20,96 +18,181 @@ public void tearDown() {
     }
 
     @Test
-    public void testCount1() {
+    //first test
+    public void count_CountingLength_ReturnCorrectLength() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
     @Test
-    public void testCount2() {
-        fail("Not yet implemented");
+    //second test
+    public void count_EmptyString_return0() {
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    @Test
+    //third test
+    public void count_nullString_return0() {
+        manipulatedstring.setString(null);
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    @Test
+    //fourth test
+    public void count_noWords_return0() {
+        manipulatedstring.setString(" ");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
     @Test
-    public void testCount3() {
-        fail("Not yet implemented");
+    //fifth test
+    public void count_trailingSpace_returnsCorrectCount(){
+        manipulatedstring.setString("Hello World! ");
+        int length = manipulatedstring.count();
+        assertEquals(2, length);
     }
 
     @Test
-    public void testCount4() {
-        fail("Not yet implemented");
+    //sixth test
+    public void count_leadingSpace_returnsCorrectCount(){
```

Comment:
this test seems a little redundant since the test above already tests if a space might accidently count another word.